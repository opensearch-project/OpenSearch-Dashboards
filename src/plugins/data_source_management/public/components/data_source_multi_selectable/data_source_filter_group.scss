.dataSourceFilterGroupItems {
  overflow: scroll;
  max-width: 300px;

  .dataSourceFilterGroupOuiFlexGroup {
    .dataSourceFilterGroupOuiFlexItem {
      text-overflow: ellipsis;
      overflow: hidden;
      white-space: nowrap;
      display: inline-block;
    }
  }
}
