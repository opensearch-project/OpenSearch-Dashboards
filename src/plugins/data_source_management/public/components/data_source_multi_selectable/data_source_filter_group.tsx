/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { useState } from 'react';
import {
  EuiNotificationBadge,
  EuiFilterSelectItem,
  EuiPopover,
  EuiPopoverTitle,
  EuiFieldSearch,
  FilterChecked,
  EuiPopoverFooter,
  EuiButtonGroup,
  EuiButtonEmpty,
} from '@elastic/eui';
import { DataSourceOption } from '../data_source_selector/data_source_selector';

export interface SelectedDataSourceOption extends DataSourceOption {
  label: string;
  id: string;
  visible: boolean;
  checked?: FilterChecked;
}

export interface DataSourceFilterGroupProps {
  selectedOptions: SelectedDataSourceOption[];
  setSelectedOptions: (options: SelectedDataSourceOption[]) => void;
}

type SelectionToggleOptionIds = 'select_all' | 'deselect_all';

const selectionToggleButtons = [
  {
    id: 'select_all',
    label: 'Select all',
  },
  {
    id: 'deselect_all',
    label: 'Deselect all',
  },
];

export const DataSourceFilterGroup: React.FC<DataSourceFilterGroupProps> = ({
  selectedOptions,
  setSelectedOptions,
}) => {
  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
  const [selectionToggleSelectedId, setSelectionToggleSelectedId] = useState<
    SelectionToggleOptionIds
  >('select_all');

  const onButtonClick = () => {
    setIsPopoverOpen(!isPopoverOpen);
  };

  const closePopover = () => {
    setIsPopoverOpen(false);
  };

  function toggleItem(index: number) {
    if (!selectedOptions[index]) {
      return;
    }

    const newItems = [...selectedOptions];

    if (newItems[index].checked === 'on') {
      newItems[index] = {
        ...newItems[index],
        checked: undefined,
      };
    } else {
      newItems[index] = {
        ...newItems[index],
        checked: 'on',
      };
    }

    setSelectedOptions(newItems);
  }

  function onSelectionToggleChange(optionId: string) {
    setSelectionToggleSelectedId(optionId as SelectionToggleOptionIds);
    toggleAll(optionId === 'select_all' ? 'on' : undefined);
  }

  function toggleAll(state: 'on' | undefined) {
    const optionsAfterToggle = selectedOptions.map((option) => ({
      ...option,
      checked: state,
    }));

    setSelectedOptions(optionsAfterToggle);
  }

  function search(term: string) {
    const optionsAfterSearch = selectedOptions.map((option) => {
      option.visible = option.label.toLowerCase().includes(term.toLowerCase());
      return option;
    });
    setSelectedOptions(optionsAfterSearch);
  }

  const numActiveSelections = selectedOptions.filter((option) => option.checked === 'on').length;
  const button = (
    <>
      <EuiButtonEmpty
        className="euiHeaderLink"
        onClick={onButtonClick}
        iconType="database"
        size="s"
        data-test-subj="dataSourceFilterGroupButton"
      >
        {'Data sources'}
      </EuiButtonEmpty>
      <EuiNotificationBadge color="subdued">{numActiveSelections}</EuiNotificationBadge>
    </>
  );

  return (
    <EuiPopover
      id="popoverExampleMultiSelect"
      button={button}
      isOpen={isPopoverOpen}
      closePopover={closePopover}
      panelPaddingSize="none"
    >
      <EuiPopoverTitle paddingSize="s">
        <EuiFieldSearch
          compressed
          onSearch={search}
          isClearable={true}
          data-test-subj="dataSourceMultiSelectFieldSearch"
        />
      </EuiPopoverTitle>
      <div className="ouiFilterSelect__items" style={{ maxHeight: 400, overflow: 'scroll' }}>
        {selectedOptions.map((item, index) => {
          const itemStyle: any = {};
          itemStyle.display = !item.visible ? 'none' : itemStyle.display;

          return (
            <EuiFilterSelectItem
              checked={item.checked}
              key={index}
              onClick={() => toggleItem(index)}
              showIcons={true}
              style={itemStyle}
            >
              {item.label}
            </EuiFilterSelectItem>
          );
        })}
      </div>
      <EuiPopoverFooter>
        <EuiButtonGroup
          legend="All options selection toggle group"
          options={selectionToggleButtons}
          idSelected={selectionToggleSelectedId}
          onChange={onSelectionToggleChange}
        />
      </EuiPopoverFooter>
    </EuiPopover>
  );
};
