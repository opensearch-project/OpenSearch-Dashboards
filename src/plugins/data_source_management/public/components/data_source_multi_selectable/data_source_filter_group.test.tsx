/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { ShallowWrapper, shallow } from 'enzyme';
import React from 'react';
import { DataSourceFilterGroup } from './data_source_filter_group';
import { render, fireEvent, screen } from '@testing-library/react';

describe('DataSourceFilterGroup', () => {
  let component: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

  it('should render normally', () => {
    const mockCallBack = jest.fn();
    component = shallow(
      <DataSourceFilterGroup
        selectedOptions={[{ id: '1', label: 'name1', checked: 'on', visible: true }]}
        setSelectedOptions={(items) => mockCallBack(items)}
        defaultDataSource="1"
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render popup when clicking on button', async () => {
    const mockCallBack = jest.fn();
    const container = render(
      <DataSourceFilterGroup
        selectedOptions={[{ id: '1', label: 'name1', checked: 'on', visible: true }]}
        setSelectedOptions={(items) => mockCallBack(items)}
        defaultDataSource="1"
      />
    );
    const button = await container.findByTestId('dataSourceFilterGroupButton');
    button.click();
    expect(container).toMatchSnapshot();
    expect(mockCallBack).toBeCalledTimes(0);

    fireEvent.click(screen.getByText('name1'));
    expect(mockCallBack).toBeCalledWith([
      { checked: undefined, id: '1', label: 'name1', visible: true },
    ]);
  });

  it('should toggle all when clicking on button and should search', async () => {
    const mockCallBack = jest.fn();
    const container = render(
      <DataSourceFilterGroup
        selectedOptions={[{ id: '1', label: 'name1', checked: 'on', visible: true }]}
        setSelectedOptions={(items) => mockCallBack(items)}
      />
    );
    const button = await container.findByTestId('dataSourceFilterGroupButton');
    button.click();

    fireEvent.click(screen.getByText('Deselect all'));
    expect(mockCallBack).toBeCalledWith([
      { checked: undefined, id: '1', label: 'name1', visible: true },
    ]);

    fireEvent.click(screen.getByText('Select all'));
    expect(mockCallBack).toBeCalledWith([
      { checked: 'on', id: '1', label: 'name1', visible: true },
    ]);

    const input = screen.getByTestId('dataSourceMultiSelectFieldSearch');
    fireEvent.change(input, { target: { value: 'random input' } });
    fireEvent.keyDown(input, { key: 'enter', keyCode: 13 });

    expect(mockCallBack).toBeCalledWith([
      { checked: 'on', id: '1', label: 'name1', visible: false },
    ]);

    expect(container).toMatchSnapshot();
  });
});
