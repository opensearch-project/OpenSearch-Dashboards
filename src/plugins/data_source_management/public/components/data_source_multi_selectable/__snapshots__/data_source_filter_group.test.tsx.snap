// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataSourceFilterGroup should render normally 1`] = `
<EuiPopover
  anchorPosition="downCenter"
  button={
    <React.Fragment>
      <EuiButtonEmpty
        className="euiHeaderLink"
        data-test-subj="dataSourceFilterGroupButton"
        iconType="database"
        onClick={[Function]}
        size="s"
      >
        Data sources
      </EuiButtonEmpty>
      <EuiNotificationBadge
        color="subdued"
      >
        1
      </EuiNotificationBadge>
    </React.Fragment>
  }
  closePopover={[Function]}
  display="inlineBlock"
  hasArrow={true}
  id="popoverExampleMultiSelect"
  isOpen={false}
  ownFocus={true}
  panelPaddingSize="none"
>
  <EuiPopoverTitle
    paddingSize="s"
  >
    <EuiFieldSearch
      compressed={true}
      data-test-subj="dataSourceMultiSelectFieldSearch"
      fullWidth={false}
      incremental={false}
      isClearable={true}
      isLoading={false}
      onSearch={[Function]}
    />
  </EuiPopoverTitle>
  <div
    className="ouiFilterSelect__items"
    style={
      Object {
        "maxHeight": 400,
        "overflow": "scroll",
      }
    }
  >
    <EuiFilterSelectItem
      checked="on"
      key="0"
      onClick={[Function]}
      showIcons={true}
      style={
        Object {
          "display": undefined,
        }
      }
    >
      <DataSourceOptionItem
        defaultDataSource="1"
        item={
          Object {
            "checked": "on",
            "id": "1",
            "label": "name1",
            "visible": true,
          }
        }
      />
    </EuiFilterSelectItem>
  </div>
  <EuiPopoverFooter>
    <EuiButtonGroup
      idSelected="select_all"
      legend="All options selection toggle group"
      onChange={[Function]}
      options={
        Array [
          Object {
            "id": "select_all",
            "label": "Select all",
          },
          Object {
            "id": "deselect_all",
            "label": "Deselect all",
          },
        ]
      }
    />
  </EuiPopoverFooter>
</EuiPopover>
`;

exports[`DataSourceFilterGroup should render popup when clicking on button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="euiPopover euiPopover--anchorDownCenter"
        id="popoverExampleMultiSelect"
      >
        <div
          class="euiPopover__anchor"
        >
          <button
            class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--small euiHeaderLink"
            data-test-subj="dataSourceFilterGroupButton"
            type="button"
          >
            <span
              class="euiButtonContent euiButtonEmpty__content"
            >
              <span
                class="euiButtonContent__icon"
                color="inherit"
                data-euiicon-type="database"
              />
              <span
                class="euiButtonEmpty__text"
              >
                Data sources
              </span>
            </span>
          </button>
          <span
            class="euiNotificationBadge euiNotificationBadge--subdued"
          >
            1
          </span>
        </div>
      </div>
    </div>
    <div>
      <div
        data-eui="EuiFocusTrap"
      >
        <div
          aria-describedby="generated-id"
          aria-live="off"
          aria-modal="true"
          class="euiPanel euiPanel--borderRadiusMedium euiPanel--plain euiPanel--noShadow euiPopover__panel euiPopover__panel--bottom"
          role="dialog"
          style="top: 16px; left: -22px; will-change: transform, opacity; z-index: 2000;"
          tabindex="0"
        >
          <div
            class="euiPopover__panelArrow euiPopover__panelArrow--bottom"
            style="left: 10px; top: 0px;"
          />
          <p
            class="euiScreenReaderOnly"
            id="generated-id"
          >
            You are in a dialog. To close this dialog, hit escape.
          </p>
          <div>
            <div
              class="euiPopoverTitle euiPopoverTitle--paddingSmall"
            >
              <div
                class="euiFormControlLayout euiFormControlLayout--compressed"
              >
                <div
                  class="euiFormControlLayout__childrenWrapper"
                >
                  <input
                    class="euiFieldSearch euiFieldSearch--compressed"
                    data-test-subj="dataSourceMultiSelectFieldSearch"
                    type="search"
                  />
                  <div
                    class="euiFormControlLayoutIcons"
                  >
                    <span
                      class="euiFormControlLayoutCustomIcon"
                    >
                      <span
                        aria-hidden="true"
                        class="euiFormControlLayoutCustomIcon__icon"
                        data-euiicon-type="search"
                      />
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ouiFilterSelect__items"
              style="max-height: 400px; overflow: scroll;"
            >
              <button
                class="euiFilterSelectItem"
                role="option"
                type="button"
              >
                <span
                  class="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow"
                >
                  <div
                    class="euiFlexItem euiFlexItem--flexGrowZero"
                  >
                    <span
                      color="text"
                      data-euiicon-type="check"
                    />
                  </div>
                  <span
                    class="euiFlexItem euiFilterSelectItem__content"
                  >
                    <div
                      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive"
                    >
                      <div
                        class="euiFlexItem euiFlexItem--flexGrow1"
                      >
                        name1
                      </div>
                      <div
                        class="euiFlexItem euiFlexItem--flexGrowZero"
                      >
                        <span
                          class="euiBadge euiBadge--iconLeft"
                          style="background-color: rgb(211, 218, 230); color: rgb(0, 0, 0);"
                        >
                          <span
                            class="euiBadge__content"
                          >
                            <span
                              class="euiBadge__text"
                            >
                              Default
                            </span>
                          </span>
                        </span>
                      </div>
                    </div>
                  </span>
                </span>
              </button>
            </div>
            <div
              class="euiPopoverFooter"
            >
              <fieldset
                class="euiButtonGroup euiButtonGroup--small euiButtonGroup--text"
              >
                <legend
                  class="euiScreenReaderOnly"
                >
                  All options selection toggle group
                </legend>
                <div
                  class="euiButtonGroup__buttons"
                >
                  <label
                    class="euiButtonGroupButton euiButtonGroupButton--text euiButtonGroupButton--small euiButtonGroupButton--fill euiButtonGroupButton-isSelected"
                    for="generated-id"
                  >
                    <span
                      class="euiButtonContent euiButton__content"
                    >
                      <span
                        class="euiButton__text euiButtonGroupButton__textShift"
                      >
                        <input
                          checked=""
                          class="euiScreenReaderOnly"
                          data-test-subj="select_all"
                          id="generated-id"
                          name="generated-id"
                          type="radio"
                          value=""
                        />
                        Select all
                      </span>
                    </span>
                  </label>
                  <label
                    class="euiButtonGroupButton euiButtonGroupButton--text euiButtonGroupButton--small"
                    for="generated-id"
                  >
                    <span
                      class="euiButtonContent euiButton__content"
                    >
                      <span
                        class="euiButton__text euiButtonGroupButton__textShift"
                      >
                        <input
                          class="euiScreenReaderOnly"
                          data-test-subj="deselect_all"
                          id="generated-id"
                          name="generated-id"
                          type="radio"
                          value=""
                        />
                        Deselect all
                      </span>
                    </span>
                  </label>
                </div>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="euiPopover euiPopover--anchorDownCenter"
      id="popoverExampleMultiSelect"
    >
      <div
        class="euiPopover__anchor"
      >
        <button
          class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--small euiHeaderLink"
          data-test-subj="dataSourceFilterGroupButton"
          type="button"
        >
          <span
            class="euiButtonContent euiButtonEmpty__content"
          >
            <span
              class="euiButtonContent__icon"
              color="inherit"
              data-euiicon-type="database"
            />
            <span
              class="euiButtonEmpty__text"
            >
              Data sources
            </span>
          </span>
        </button>
        <span
          class="euiNotificationBadge euiNotificationBadge--subdued"
        >
          1
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`DataSourceFilterGroup should toggle all when clicking on button and should search 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="euiPopover euiPopover--anchorDownCenter"
        id="popoverExampleMultiSelect"
      >
        <div
          class="euiPopover__anchor"
        >
          <button
            class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--small euiHeaderLink"
            data-test-subj="dataSourceFilterGroupButton"
            type="button"
          >
            <span
              class="euiButtonContent euiButtonEmpty__content"
            >
              <span
                class="euiButtonContent__icon"
                color="inherit"
                data-euiicon-type="database"
              />
              <span
                class="euiButtonEmpty__text"
              >
                Data sources
              </span>
            </span>
          </button>
          <span
            class="euiNotificationBadge euiNotificationBadge--subdued"
          >
            1
          </span>
        </div>
      </div>
    </div>
    <div>
      <div
        data-eui="EuiFocusTrap"
      >
        <div
          aria-describedby="generated-id"
          aria-live="off"
          aria-modal="true"
          class="euiPanel euiPanel--borderRadiusMedium euiPanel--plain euiPanel--noShadow euiPopover__panel euiPopover__panel--bottom"
          role="dialog"
          style="top: 16px; left: -22px; will-change: transform, opacity; z-index: 2000;"
          tabindex="0"
        >
          <div
            class="euiPopover__panelArrow euiPopover__panelArrow--bottom"
            style="left: 10px; top: 0px;"
          />
          <p
            class="euiScreenReaderOnly"
            id="generated-id"
          >
            You are in a dialog. To close this dialog, hit escape.
          </p>
          <div>
            <div
              class="euiPopoverTitle euiPopoverTitle--paddingSmall"
            >
              <div
                class="euiFormControlLayout euiFormControlLayout--compressed"
              >
                <div
                  class="euiFormControlLayout__childrenWrapper"
                >
                  <input
                    class="euiFieldSearch euiFieldSearch--compressed euiFieldSearch-isClearable"
                    data-test-subj="dataSourceMultiSelectFieldSearch"
                    type="search"
                  />
                  <div
                    class="euiFormControlLayoutIcons"
                  >
                    <span
                      class="euiFormControlLayoutCustomIcon"
                    >
                      <span
                        aria-hidden="true"
                        class="euiFormControlLayoutCustomIcon__icon"
                        data-euiicon-type="search"
                      />
                    </span>
                  </div>
                  <div
                    class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
                  >
                    <button
                      aria-label="Clear input"
                      class="euiFormControlLayoutClearButton euiFormControlLayoutClearButton--small"
                      type="button"
                    >
                      <span
                        class="euiFormControlLayoutClearButton__icon"
                        data-euiicon-type="cross"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ouiFilterSelect__items"
              style="max-height: 400px; overflow: scroll;"
            >
              <button
                class="euiFilterSelectItem"
                role="option"
                type="button"
              >
                <span
                  class="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow"
                >
                  <div
                    class="euiFlexItem euiFlexItem--flexGrowZero"
                  >
                    <span
                      color="text"
                      data-euiicon-type="check"
                    />
                  </div>
                  <span
                    class="euiFlexItem euiFilterSelectItem__content"
                  >
                    <div
                      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive"
                    >
                      <div
                        class="euiFlexItem euiFlexItem--flexGrow1"
                      >
                        name1
                      </div>
                    </div>
                  </span>
                </span>
              </button>
            </div>
            <div
              class="euiPopoverFooter"
            >
              <fieldset
                class="euiButtonGroup euiButtonGroup--small euiButtonGroup--text"
              >
                <legend
                  class="euiScreenReaderOnly"
                >
                  All options selection toggle group
                </legend>
                <div
                  class="euiButtonGroup__buttons"
                >
                  <label
                    class="euiButtonGroupButton euiButtonGroupButton--text euiButtonGroupButton--small euiButtonGroupButton--fill euiButtonGroupButton-isSelected"
                    for="generated-id"
                  >
                    <span
                      class="euiButtonContent euiButton__content"
                    >
                      <span
                        class="euiButton__text euiButtonGroupButton__textShift"
                        data-text="Select all"
                        title="Select all"
                      >
                        <input
                          checked=""
                          class="euiScreenReaderOnly"
                          data-test-subj="select_all"
                          id="generated-id"
                          name="generated-id"
                          type="radio"
                          value=""
                        />
                        Select all
                      </span>
                    </span>
                  </label>
                  <label
                    class="euiButtonGroupButton euiButtonGroupButton--text euiButtonGroupButton--small"
                    for="generated-id"
                  >
                    <span
                      class="euiButtonContent euiButton__content"
                    >
                      <span
                        class="euiButton__text euiButtonGroupButton__textShift"
                        data-text="Deselect all"
                        title="Deselect all"
                      >
                        <input
                          class="euiScreenReaderOnly"
                          data-test-subj="deselect_all"
                          id="generated-id"
                          name="generated-id"
                          type="radio"
                          value=""
                        />
                        Deselect all
                      </span>
                    </span>
                  </label>
                </div>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="euiPopover euiPopover--anchorDownCenter"
      id="popoverExampleMultiSelect"
    >
      <div
        class="euiPopover__anchor"
      >
        <button
          class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--small euiHeaderLink"
          data-test-subj="dataSourceFilterGroupButton"
          type="button"
        >
          <span
            class="euiButtonContent euiButtonEmpty__content"
          >
            <span
              class="euiButtonContent__icon"
              color="inherit"
              data-euiicon-type="database"
            />
            <span
              class="euiButtonEmpty__text"
            >
              Data sources
            </span>
          </span>
        </button>
        <span
          class="euiNotificationBadge euiNotificationBadge--subdued"
        >
          1
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
